src/components/FeaturePanel/PublicTransport/PublicTransport.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { requestLines } from './requestRoutes';
+import { PublicTransportWrapper } from './PublicTransportWrapper';
+import { FeatureTags } from '../../../services/types';
+import { LineNumber } from './LineNumber';
+
+interface PublicTransportProps {
+  tags: FeatureTags;
+}
+
+export const PublicTransport: React.FC<PublicTransportProps> = ({ tags }) => {
+  const isPublicTransport =
+    Object.keys(tags).includes('public_transport') ||
+    tags.railway === 'station';
+
+  if (!isPublicTransport) {
+    return null;
+  }
+
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const router = useRouter();
+
+  useEffect(() => {
+    const loadData = async () => {
+      const lines = await requestLines(
+        router.query.all[0] as any,
+        Number(router.query.all[1]),
+      );
+      setData(lines);
+      setLoading(false);
+    };
+
+    loadData();
```

Comment:
Would you be able to treat the error state please? If there is no connection or overpass throws "http error 500", it would leave the loading state forever.

You can take inspiration [here](https://github.com/zbycz/osmapp/blob/master/src/components/FeaturePanel/ObjectsAround.tsx#L72-L73), or perhaps add react-query library if you know it already.